R/learner_mgcv_regr_gam.R

```diff
@@ -0,0 +1,195 @@
+#' @title Regression Generalized Additive Model Learner
+#' @author JazzyPierrot
+#' @name mlr_learners_regr.gam
+#'
+#' @description
+#' Generalized additive models.
+#' Calls [mgcv::gam] from package \CRANpkg{mgcv}.
+#'
+#' A gam formula specific to the task at hand is required for the `formula`
+#' parameter (see example and `?mgcv::formula.gam`). Beware, if no formula is provided, a fallback formula is
+#' used that will make the gam behave like a glm (this behavior is required
+#' for the unit tests). Only features specified in the formula will be used,
+#' superseding columns with col_roles "feature" in the task.
+#'
+#'
+#' @template class_learner
+#' @templateVar id regr.gam
+#' @templateVar caller gam
+#'
+#' @references
+#'
+#' Wood, S.N. (2017) Generalized Additive Models: An Introduction with R (2nd edition). Chapman
+#' & Hall/ CRC, Boca Raton, Florida
+#'
+#' Key Reference on GAMs generally:
+#'
+#' Hastie (1993) in Chambers and Hastie (1993) Statistical Models in S. Chapman
+#' and Hall.
+#'
+#' Hastie and Tibshirani (1990) Generalized Additive Models. Chapman and Hall.
+#'
+#' @template seealso_learner
+#' @template example
+#'
+#' @examples
+#'
+#' # simple example
+#' t = mlr3::tsk("mtcars")
+#' l = mlr3::lrn("regr.gam")
+#' l$param_set$values = list(
+#'   formula = mpg ~ cyl + am + s(disp) + s(hp)
+#' )
```

Comment:
Sorry to be pedantic but please change to:

```suggestion
l$param_set$values$formula = mpg ~ cyl + am + s(disp) + s(hp)
```

Otherwise you're actually overwriting any other `values` (which doesn't matter in the example but does in general)

Reply:
Nothing to be sorry about, you are perfectly right !